### ISSUE TITLE
Lollypop disappeared

### ISSUE THREAD

**Issue by d9h02f** [gnumdk/lollypop#1148]
Fedora 25, I was using Lollypop 0.9.240, just updated to 0.9.243 and Lollypop has disappeared from the application overview. If I try starting it from terminal, I get "bash: /usr/bin/lollypop: Access denied". Removing/reinstalling doesn't help.

**Comment by gnumdk** [comment]
No idea, porting rpm package to meson was a hard task and looks like a fail :-1:

**Comment by gnumdk** [comment]
Please ping Fedora maintainer to update his lollypop package (and migrate to meson), then I will sync with it spec file.

**Comment by badshah400** [comment]
I maintain lollypop for openSUSE. I noticed that, after the port to meson, one needs to manually set exec bits on the lollypop binaries.

@d9h02f 
Does doing `sudo chmod +x /usr/bin/lollypop /usr/bin/lollpop-cli /usr/bin/rhythmbox2lollypop` help?

**Comment by d9h02f** [comment]
Thanks, that solved the problem on my side. I don't know how to contact the package maintainer for Fedora, so I'll just wait until he fixes it.

**Comment by cworobetz** [comment]
This issue is also affecting me on Debian 9, as downloaded from the Ubuntu PPA.

**Comment by gnumdk** [comment]
@badshah400 Any idea on what is happening? +x bit is set by an external script in meson build. Looks like it's not executed when building rpm/deb  :-1: 

@bil-elmoussaoui Any idea?

**Comment by badshah400** [comment]
@gnumdk My hunch is that typically the rpm specfile exports the variable DESTDIR when running `ninja` for the `%install` stage. For example
`DESTDIR=%{buildroot} %{__ninja} install %{__ninja_common_opts}`
is how openSUSE calls ninja for the `%install` stage.

Your post_install script does this:

```python
destdir = environ.get('DESTDIR', '')

if not destdir:
    print('Updating icon cache...')
    call(['gtk-update-icon-cache', '-qtf', path.join(datadir, 'icons', 'hicolor')])
    print("Installing new Schemas")
    call(['glib-compile-schemas', path.join(datadir, 'glib-2.0/schemas')])
    call(['chmod', '+x', path.join(prefix, 'bin', 'lollypop')])
    call(['chmod', '+x', path.join(prefix, 'bin', 'lollypop-cli')])
    call(['chmod', '+x', path.join(prefix, 'libexec', 'lollypop-sp')])
```

This means when DESTDIR is set, the chmod calls (and others in the same if block) are not executed. At least, that is what I understand.

**Comment by gnumdk** [comment]
@bil-elmoussaoui You confirm the issue?

**Comment by bil-elmoussaoui** [comment]
@gnumdk We can fix that differently by setting the files executable by default

**Comment by bil-elmoussaoui** [comment]
@gnumdk This can be closed too

**Comment by gnumdk** [comment]
Initial bug is about my packages, will close as soon I release a fix (tomorrow I hope)

**Comment by repsorp** [comment]
Lollypop totally disappeared with last update
lollypop/xenial,xenial 0.9.244-1ppa1ubuntu1.1 all [upgradable from: 0.9.242-1ppa1ubuntu1.1]

Not more Lollypop file in /usr/bin

Ubuntu 16.04

**Comment by JamesTiberiusKirk** [comment]
@repsorp I have the same problem in Elementary Os 0.4.1 Loki using the ubuntu PPA

**Comment by Pointedstick** [comment]
Yeah  0.9.244-1ppa1ubuntu1.1 deleted the app entirely.

**Comment by gnumdk** [comment]
Help welcome, Don't really know what is happening...

Fedora user? All is ok with package?

**Comment by gnumdk** [comment]
OpenSUSE user: broken package too, I know :-1:

**Comment by badshah400** [comment]
@gnumdk openSUSE package (v0.9.244) from [obs://multimedia:apps](https://build.opensuse.org/package/show/multimedia:apps/lollypop) is working a-ok for me. Where are you getting your pkg from?

**Comment by gnumdk** [comment]
Cool :) Will take openSUSE spec file.

Are you sure? Here https://build.opensuse.org/package/show/openSUSE%3AFactory/lollypop 0.9.240 :-(

**Comment by badshah400** [comment]
Update to Factory is pending review: https://build.opensuse.org/request/show/509585
But, like I said in my previous comment (geez! GH should implement comment numbering!), you can also just get the spec file from [multimedia:apps](https://build.opensuse.org/package/show/multimedia:apps/lollypop) which is where updates to Factory are submitted from.

**Comment by gnumdk** [comment]
Any Ubuntu/Debian package in the place?

**Comment by marius-wieschollek** [comment]
@gnumdk I'm getting it from the official Ubuntu PPA. There is no content at all in the package besides the changelog.

**Comment by gnumdk** [comment]
I know...

**Comment by mray** [comment]
For the record, Ubuntu 17.04 is also affected.

**Comment by Pointedstick** [comment]
If this is a tricky problem to fix, I would recommend backing out the changes. A lot of people got your app totally deleted, with often no way to get it back (PPAs don't keep the previous version around).

**Comment by gnumdk** [comment]
\o/ FIXED \o/

**Comment by Pointedstick** [comment]
Can you push out an update to the Ubuntu PPA?

**Comment by GuilloOme** [comment]
@Pointedstick It has been updated… you probably have to uninstall then reinstall it to make it works again.

**Comment by vegbrasil** [comment]
@Pointedstick `sudo apt update` and `sudo apt upgrade` solved for me in Ubuntu 17.04.

**Comment by Pointedstick** [comment]
Yay, thanks!
